pandas/tests/dtypes/test_inference.py

```diff
@@ -618,6 +618,30 @@ def test_decimals(self):
         result = lib.infer_dtype(arr, skipna=True)
         assert result == 'decimal'
 
+    def test_complex(self):
+        # gets cast to complex on array construction
+        arr = np.array([1.0, 2.0, 1 + 1j])
+        result = lib.infer_dtype(arr, skipna=True)
```

Comment:
does skipna=False matter anywhere here?

Reply:
No, because `np.nan` is compatible with complex. The only thing is that not passing `skipna` yields a warning.

Reply:
the parametrize over skipna